src/main/java/com/google/cloud/tools/appengine/AppEngineDescriptor.java

```diff
@@ -72,12 +72,15 @@ public String getProjectId() throws AppEngineException {
   }
 
   /**
-   * Returns runtime from the &lt;runtime&gt; element of the appengine-web.xml or null if it is
-   * missing.
+   * Returns runtime from the &lt;runtime&gt; element of the appengine-web.xml or the default one
+   * when it is missing.
    */
-  @Nullable
   public String getRuntime() throws AppEngineException {
-    return getText(getNode(document, "appengine-web-app", "runtime"));
+    String runtime = getText(getNode(document, "appengine-web-app", "runtime"));
+    if (runtime == null) {
```

Comment:
Should we start failing java7 builds when that happens?

Reply:
When java7 is decommissioned, the admin deployment backend will emit an error like any other unknown runtime... It would be nice if local tooling would detect this situation before deployment and during development (execution or pre stage).  This cannot be done before Q1 next year.

Reply:
For now, it is still a valid configuration (no defined runtime is defaulted to java7). In the future, it means that the runtime field will become mandatory.

Reply:
So if there's no runtime element in appengine-web.xml this should still be treated as java7 at that point in time?

Reply:
I believe that will result in an error at that time.? @ludoch ?

Reply:
This has always been the case: no runtime entry is equivalent to runtime java7.

Reply:
In Q1, no runtime in appengine-web.xml should be treated as an error.